qiskit/circuit/qpy_serialization.py

```diff
@@ -340,6 +387,11 @@
 the circuit's qubit index. For example, array element 0's value is the index
 of the ``register[0]``'s position in the containing circuit's qubits list.
 
+.. note::
+
+    Starting in QPY Version 4 the type of array elements has changed to int64_t
+    and a negative value represents a bit in the array not present in the circuit
+
```

Comment:
It might be clearer in the documentation if we change the main text to say `int64_t`, and make the note say "prior to version 4, the type of array elements was `uint32_t`".

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7584/commits/a627ef4ec357708aec02a5263b845de8d6a314bf